### PR TITLE
Add Vector_Vector_Activate_Stream_Batch from umav1511

### PR DESCRIPTION
**Pull request by maltanar**
Credit: @umav1511

Add VVAU variant with decoupled weight stream.

Originally from https://github.com/umav1511/finn-hlslib/blob/e22a73b462a99a5b4d5bdbb7ade0f2e7b4ca8e0f/vvau.hpp#L189 but also changed pipeline pragma to use the flushable style for Vitis HLS.

### CODE REVIEW

File: vvau.hpp [hunk 1]
```diff
@@ -53,6 +53,113 @@
...
+	R const &r
+) {
+
+	// how many different rows each neuron will compute
+	// alternatively: number of vertical matrix chunks
+	constexpr unsigned  NF = Channels / PE;
+
+	// how many synapse groups each row is split into
+	// alternatively: number of horizontal matrix chunks
+	constexpr unsigned  SF = (Channels*Kernel_2) / Channels;
```
Comment (preusser) [inline vvau.hpp:101]: What is the rationale for going forth and then back again?

**Review by preusser — CHANGES_REQUESTED** [review]

File: vvau.hpp [hunk 2]
```diff
@@ -53,6 +53,113 @@
 /**
  * \brief Vector vector activate function
  *
+ * The function performs the multiplication between a weight vector and the input activation vector,
+ * accumulating the results and then applying an activation function on the accumulated result.
+ * It is used to implement depth-wise separable convolution
+ * 
+ * \tparam Channels   Number of channels
```
[the remark this answers is not in the document]
Comment (maltanar) [reply vvau.hpp:59]: VVAU currently only implement "standard" depthwise convolutions with input channels = output channels, so I'd say it's fair to call this just "channels". Same convention is applied in the FINN compiler too: https://github.com/Xilinx/finn/blob/main/src/finn/custom_op/fpgadataflow/vector_vector_activate_batch.py#L58

File: vvau.hpp [hunk 3]
```diff
@@ -53,6 +53,113 @@
...
+ * \tparam Kernel_2   Kernel * Kernel dimension (Kernel ^ 2 if square)
+ * \tparam SIMD       Number of input columns computed in parallel
```
[the remark this answers is not in the document]
Comment (maltanar) [reply vvau.hpp]: Added a `static_assert` to ensure `SIMD==1`.

File: vvau.hpp (hunk 1 above)
Comment (maltanar) [reply vvau.hpp]: I don't think there is any good rationale, perhaps SIMD was part of the equation at some point and was partially simplifed.

**Review by preusser — APPROVED** [review]

**Pull request merged**